tests/core/test_task.py

```diff
@@ -92,6 +92,19 @@ def test_create_task_illegal_handler(self):
         with pytest.raises(TypeError):
             Task(state_handlers=lambda *a: 1)
 
+    def test_class_creation_rejects_reserved_init_kwargs(self):
+        with pytest.raises(ValueError):
+
+            class BadTask(Task):
+                def __init__(self, name):
+                    pass
+
+        with pytest.raises(ValueError):
+
+            class BadTask(Task):
```

Comment:
Does this get applied on subclasses-of-subclasses? Like if someone subclassed FunctionTask, would that subclass be validated?

Reply:
Unfortunately... kidding - it does get applied to subclasses if subclasses!